patches/server/0905-Reapply-game-event-listener-on-update.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Wed, 8 Jun 2022 22:49:31 -0400
+Subject: [PATCH] Reapply game event listener on update
+
+Some block entities update their listener when load is called. This is bad as then when the block entity is
+removed the listener will not be able to be removed. This properly removes and readds listeners when updating
+block entities that have this special behavior.
+
+diff --git a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
+index 9cce218623a71955687817a6317ed519f3300368..c7aa874a636f18072a4b0c373cc0573cf43ee75b 100644
+--- a/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
++++ b/src/main/java/net/minecraft/world/level/chunk/LevelChunk.java
+@@ -757,7 +757,7 @@ public class LevelChunk extends ChunkAccess {
+         this.removeBlockEntityTicker(pos);
+     }
+ 
+-    private <T extends BlockEntity> void removeGameEventListener(T blockEntity, ServerLevel worldserver) {
++    public  <T extends BlockEntity> void removeGameEventListener(T blockEntity, ServerLevel worldserver) { // Paper
```

Comment:
Are ATs still used or are we slowly moving away from them?

Reply:
I'd like feedback from other developers, but from my interpretation yes, we appear to be moving away from them.

I'm personally trying to only use them with stuff like fields, where it's just easy to make them not final and Publix.

Reply:
General gist: Use ATs for methods that are overridden and you need to cover a whole hierarchy or in high-conflict areas, otherwise just patching one method or field is easier